PAB-4691 - Change log entry for duplicate alarms with geofence smart rule (#2857)
SoftwareAG/c8y-docs: content/change-logs/analytics/apama-in-c8y-20250128-The-Smartrule-service-will-now-disregard-rule-edits-unless-there-is-a-configuration-change.md

---
date:

> title: Geofence smart rule no longer creates duplicate alarms.

Reviewer: ```suggestion
title: Geofence smart rule no longer creates duplicate alarms
```

> title: The geofence smart rule is not generating duplicate alarms.

Reviewer: ```suggestion
title: Geofence smart rule no longer creates duplicate alarms
```

change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
product_area: Analytics
component:
  - value: component-M5-cepIIS
    label: Streaming Analytics
build_artifact:
  - value: tc-KXXmo2SUR
    label: apama-in-c8y
ticket: PAB-4691
version: 26.27.0
---

> The **On geofence create alarm** smart rule was periodically creating duplicate alarms if the geofence configuration is triggered. This issue has now been resolved.

Reviewer: ```suggestion
The **On geofence create alarm** smart rule was periodically creating duplicate alarms when the geofence configuration was triggered. This issue has now been resolved.
```

---

> date: 2025-01-28

Reviewer 2: Please leave the date field as empty. This will be populated on the date this build is deployed to eu-latest

Author: done [56099dd](https://github.com/Cumulocity-IoT/c8y-docs/pull/2857/commits/56099dd672f64e95c5e33529ff13a6b007d4a05c)

> title: The Smart rule service will now disregard rule edits unless there is a configuration change

Reviewer 2: I don't believe this is appropriate here. This is just implementation detail.

Author: done

change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
product_area: Analytics
component:
  - value: component-M5-cepIIS
    label: Streaming Analytics
build_artifact:
  - value: tc-KXXmo2SUR
    label: apama-in-c8y
ticket: PAB-4691
version: 26.27.0
---

> The Smart rule service will now disregard rule edits unless there is a configuration change. 

Reviewer 2: delete, just a repetition of title

Author: removed

> Two issues have been fixed:  

Reviewer 2: delete

Author: done

> 1. The **On geofence create alarm** rule was triggering duplicate alarms after 24 hours, on position updates despite no changes in it. This issue has now been resolved.  

Reviewer 2: How about something like?
```suggestion
The **On geofence create alarm** smart rule was periodically creating duplicate alarms if the geofence configuration is triggered. This issue has now been resolved.  
```

Author: added

> 2. The **threshold** rule was not functioning correctly when only KPIs (mainly fragments and/or series) were updated dynamically. This has also been corrected.

Reviewer 2: This has nothing to do with this change, please remove it

Author: done
